File: app/src/main/java/com/kidozh/discuzhub/results/ThreadCommentsResult.java
package com.kidozh.discuzhub.results;

import com.fasterxml.jackson.annotation.JsonFormat;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.kidozh.discuzhub.entities.ThreadInfo;

import java.util.ArrayList;
import java.util.List;

@JsonIgnoreProperties(ignoreUnknown = true)
public class ThreadCommentsResult extends BaseResult {
    @JsonProperty("Variables")
    public CommentsVariables commentsVariables;

    @JsonIgnoreProperties(ignoreUnknown = true)
    public static class CommentsVariables extends VariableResults{

        @JsonProperty("data")
        public List<ThreadInfo> forumThreadList = new ArrayList<>();
        @JsonFormat(shape = JsonFormat.Shape.STRING)
        @JsonProperty("perpage")
        public int perPage;

    }

    public boolean isError(){
        return this.message == null;
    }
}

File: app/src/main/java/com/kidozh/discuzhub/results/ThreadCommentsResult.kt
package com.kidozh.discuzhub.results

import com.fasterxml.jackson.annotation.JsonFormat
import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import com.fasterxml.jackson.annotation.JsonProperty
import com.kidozh.discuzhub.entities.Thread

@JsonIgnoreProperties(ignoreUnknown = true)
class ThreadCommentsResult : BaseResult() {
    @JsonProperty("Variables")
    var commentsVariables: CommentsVariables = CommentsVariables()

    @JsonIgnoreProperties(ignoreUnknown = true)
    class CommentsVariables : VariableResults() {
        @JsonProperty("data")
        var forumThreadList: List<Thread> = ArrayList()

        @JsonFormat(shape = JsonFormat.Shape.STRING)
        @JsonProperty("perpage")
        var perPage = 0
    }

    override fun isError(): Boolean {
        return message == null
    }
}